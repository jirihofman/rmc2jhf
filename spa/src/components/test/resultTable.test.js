import ResultTable from '../resultTable';
import React from 'react';
import { shallow } from 'enzyme';

describe('Result table component', () => {
	let props = [];
	let wrapper = null;
	beforeEach(() => {
		props = {intervals: [
			{ monthStart: 1, monthEnd: 2, amount: 1500 },
			{ monthStart: 3, monthEnd: 10, amount: 300 }
		]};
		wrapper = shallow(<ResultTable {...props} />);
	});
	it('is empty when no data', () => {
		props = null; // tabulka nedostane zadna data
		wrapper = shallow(<ResultTable {...props} />);
		expect(wrapper.get(0)).toBe(null);
	});
	it('matches snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('renderTableRow() is empty', () => {
		expect(wrapper.instance().renderTableRow()).toBe(null);
	});
});
