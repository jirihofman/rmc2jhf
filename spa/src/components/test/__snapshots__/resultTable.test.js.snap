// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result table component matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultTable
    intervals={
      Array [
        Object {
          "amount": 1500,
          "monthEnd": 2,
          "monthStart": 1,
        },
        Object {
          "amount": 300,
          "monthEnd": 10,
          "monthStart": 3,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": Array [
        <h1 />,
        Array [
          <Row
            bsClass="row"
            componentClass="div"
          >
            <div>
              1
              . - 
              2
              . 
               = 
              1500
               
            </div>
          </Row>,
          <Row
            bsClass="row"
            componentClass="div"
          >
            <div>
              3
              . - 
              10
              . 
               = 
              300
               
            </div>
          </Row>,
        ],
      ],
      "componentClass": "div",
      "fluid": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": <div>
            1
            . - 
            2
            . 
             = 
            1500
             
          </div>,
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              1,
              ". - ",
              2,
              ". ",
              undefined,
              " = ",
              1500,
              " ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            1,
            ". - ",
            2,
            ". ",
            undefined,
            " = ",
            1500,
            " ",
            undefined,
          ],
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": <div>
            3
            . - 
            10
            . 
             = 
            300
             
          </div>,
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              3,
              ". - ",
              10,
              ". ",
              undefined,
              " = ",
              300,
              " ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            3,
            ". - ",
            10,
            ". ",
            undefined,
            " = ",
            300,
            " ",
            undefined,
          ],
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": Array [
          <h1 />,
          Array [
            <Row
              bsClass="row"
              componentClass="div"
            >
              <div>
                1
                . - 
                2
                . 
                 = 
                1500
                 
              </div>
            </Row>,
            <Row
              bsClass="row"
              componentClass="div"
            >
              <div>
                3
                . - 
                10
                . 
                 = 
                300
                 
              </div>
            </Row>,
          ],
        ],
        "componentClass": "div",
        "fluid": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "bsClass": "row",
            "children": <div>
              1
              . - 
              2
              . 
               = 
              1500
               
            </div>,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                1,
                ". - ",
                2,
                ". ",
                undefined,
                " = ",
                1500,
                " ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              1,
              ". - ",
              2,
              ". ",
              undefined,
              " = ",
              1500,
              " ",
              undefined,
            ],
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "bsClass": "row",
            "children": <div>
              3
              . - 
              10
              . 
               = 
              300
               
            </div>,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                3,
                ". - ",
                10,
                ". ",
                undefined,
                " = ",
                300,
                " ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              3,
              ". - ",
              10,
              ". ",
              undefined,
              " = ",
              300,
              " ",
              undefined,
            ],
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
